### PR TITLE
parallels: simplify regex for vm files

### PR DESCRIPTION
**Pull request by lbajolet-hashicorp**
The original regex had multiple occurrences of a `.+?` pattern, which is essentially the same as a `.*`, so we replace the former by the latter for clarity.

Additionally, since the beginning of the path does not interest us, we can ignore it from the regex, and only start matching once we have found what's interesting for us, further simplifying the expression.

### CODE REVIEW

File: post-processor/vagrant/parallels.go
Comment (nywilken) [inline post-processor/vagrant/parallels.go:44]: is `pvmPath = filepath.FromSlash(matches[2])` still correct since there is only one capture would this be at index 1? 

I think it would be great to add a test for this. It wasn't clear to me what the actual file paths can be.

Comment (lbajolet-hashicorp) [reply post-processor/vagrant/parallels.go:44]: That is a good question, I tested this with the updated test that I added to #107, and it did not fail fwiw, but I'll check what happens here. I would think there's two matches only indeed, the global one and the pvm/macvm.
I'll investigate, thanks for the callout

Comment (lbajolet-hashicorp) [reply post-processor/vagrant/parallels.go:44]: Alright nevermind, I guess I didn't do my test right and it does crash now, I'll fix that now

Comment (lbajolet-hashicorp) [reply post-processor/vagrant/parallels.go:46]: Rerolled, since we don't capture the whole path now, but instead only the `.+.pvm|macvm/...`, we don't need to get a submatch and instead we can use `FindString` to return the whole path, excluding the parent directories.

Comment (lbajolet-hashicorp) [reply post-processor/vagrant/parallels.go:46]: Note: I based this PR on #107 for tests to run, we can rebase it on main before merging

Comment (nywilken) [reply post-processor/vagrant/parallels.go:46]: Thanks for making the update. I rebased onto the latest main and will merge once all goes green.

**Review by nywilken — APPROVED** [review]

**Pull request merged**
